api/rds.proto

```diff
@@ -282,6 +283,10 @@ message RouteAction {
 
   // Indicates that the route has a CORS policy.
   CorsPolicy cors = 17;
+
+  // Indicates that the route has auth checks.
+  // [V2-API-DIFF] new in v2.
+  AuthConfig auth_config = 18;
```

Comment:
Should this be in the route/redirect action or the route match?

Reply:
the RouteConfig seems to be read-only and contain only route decision information.

i think we can add a new AuthAction bisides RouteAction/RedirectAction [here](https://github.com/jwfang/data-plane-api/blob/05d16fe66d483a6c30a6248887c8a13461fcd51f/api/rds.proto#L306)

Reply:
This is fine, as long as we're not making route selection decisions based on `AuthConfig`; if we ever want to be in the situation where we select between 2+ different routes based on certificate match or SAN, then this needs to go into the `RouteMatch` message, since once we hit `*Action`, we've already picked the route and can only continue on with that action.

Reply:
i see. i think we are only doing cert verification after route selection right now and leave route selection based on certs info later.

Reply:
Makes sense. @PiotrSikora FYI.